src/wallet/coinselection.h

```diff
@@ -387,8 +371,20 @@ struct SelectionResult
     /** How much individual inputs overestimated the bump fees for shared ancestries */
     void SetBumpFeeDiscount(const CAmount discount);
 
-    /** Calculates and stores the waste for this selection via GetSelectionWaste */
-    void ComputeAndSetWaste(const CAmount min_viable_change, const CAmount change_cost, const CAmount change_fee);
+    /** Calculates and stores the waste for this result given the cost of change
+     * and the opportunity cost of spending these inputs now vs in the future.
+     * If change exists, waste = change_cost + inputs * (effective_feerate - long_term_feerate) - bump_fee_group_discount
+     * If no change, waste = excess + inputs * (effective_feerate - long_term_feerate) - bump_fee_group_discount
+     * where excess = selected_effective_value - target
+     * change_cost = effective_feerate * change_output_size + long_term_feerate * change_spend_size
+     *
+     * @param[in] min_viable_change The minimum amount necessary to make a change output economic
+     * @param[in] change_cost       The cost of creating change and spending it in the future.
+     *                              Only used if there is change, in which case it must be positive.
+     *                              Must be 0 if there is no change.
```

Comment:
From the commit message you mention we can have change_cost of 0 and also a change output, so this is no longer the case?
```suggestion
    * @param[in] change_cost       The cost of creating change and spending it in the future.
     *                              Only used if there is change.
```